packages/sdk-router/src/sdk.ts

```diff
@@ -39,141 +53,396 @@ class SynapseSDK {
+    ): Promise<any> {
+      const [bestOriginQuery, bestDestQuery, bestBridgeToken] = bestQuery;
+
+      // Set default deadlines
+      const originQuery = convertQuery(bestOriginQuery);
+      originQuery.deadline = deadline ?? TEN_MIN_DEADLINE;
+      const destQuery = convertQuery(bestDestQuery);
+      destQuery.deadline = ONE_WEEK_DEADLINE;
+
+      let feeAmount;
+      let feeConfig;
+
+      // Get fee data
+      if (isCCTP) {
+        // Cast router to SynapseCCTPRouter
+        const cctpRouter = <SynapseCCTPRouter>router;
+
+        feeAmount = await cctpRouter.routerContract.calculateFeeAmount(
+          bestBridgeToken.token,
+          originQuery.minAmountOut,
+          //TODO: how to determine isSwap?
+          false
+        );
+
+        const [relayerFee, minBaseFee, , maxFee] = await cctpRouter.routerContract.feeStructures(
+          bestBridgeToken.token
+        );
+
+        // return only the compatible fields
+        feeConfig = {
+          bridgeFee: relayerFee, // Assuming relayerFee is compatible with bridgeFee
+          minFee: minBaseFee, // assuming minBaseFee is compatible with minFee
+          maxFee // maxFee is the same in both cases
+        };
+      } else {
+        // Cast router to SynapseRouter
+        const synapseRouter = <SynapseRouter>router;
+
+        feeAmount = await synapseRouter.routerContract.calculateBridgeFee(
+          bestBridgeToken.token,
+          originQuery.minAmountOut
+        );
+
+        feeConfig = await synapseRouter.routerContract.fee(bestBridgeToken.token);
+      }
 
-    if (originQuery && destInToken) {
-      feeAmount = destRouter.routerContract.calculateBridgeFee(
-        destInToken,
-        originQuery.minAmountOut
-      )
-      feeConfig = destRouter.routerContract.fee(destInToken)
+      return {
+        feeAmount,
+        feeConfig,
+        routerAddress: router.routerContract.address,
+        maxAmountOut: destQuery.minAmountOut,
+        originQuery,
+        destQuery,
+      };
     }
 
-    return {
-      feeAmount: await feeAmount,
-      feeConfig: await feeConfig,
-      routerAddress,
-      maxAmountOut,
-      originQuery,
-      destQuery,
+
+    public async bridgeQuote(
+      originChainId: number,
+      destChainId: number,
+      tokenIn: string,
+      tokenOut: string,
+      amountIn: BigintIsh,
+      deadline?: BigNumber
+    ): Promise<BridgeQuote | undefined> {
+      tokenOut = handleNativeToken(tokenOut);
+      tokenIn = handleNativeToken(tokenIn);
+
+      const originRouters: (SynapseRouter | SynapseCCTPRouter)[] = [this.synapseRouters[originChainId], this.synapseCCTPRouters[originChainId]].filter(Boolean);
+      const destRouters: (SynapseRouter | SynapseCCTPRouter)[] = [this.synapseRouters[destChainId], this.synapseCCTPRouters[destChainId]].filter(Boolean);
+
+      let bestQuote: BridgeQuote | undefined;
+
+      for (let originRouter of originRouters) {
+        for (let destRouter of destRouters) {
```

Comment:
This for loop needs to be re-worked to not create synchronous blocking processes, increasing latency. OriginCCTPRouter can't call DestinationSynapseRouter, correct, @ChiTimesChi?

Instead, OriginCCTPRouter should only be calling DestCCTPRouter.

Reply:
Correct, every Router should only call their counterpart on the other chain.